src/ZFTool/Controller/DiagnosticsController.php

```diff
@@ -224,13 +256,78 @@ public function runAction()
             } else {
                 $model->setErrorLevel(0);
             }
-        } else {
-            // Display results as a web page
+            return $model;
+    }
+
+    private function processHttpRequest(Request $request, Collection $results)
+    {
+        $defaultAccept = new Accept();
+        $defaultAccept->addMediaType(self::CONTENT_TYPE_HTML);
+
+        $acceptHeader = $request->getHeader('Accept', $defaultAccept);
+
+        $viewModel = function () use ($results) {
             $model = new ViewModel();
             $model->setVariable('results', $results);
+            return $model;
+        };
+
+        if ($acceptHeader->match(self::CONTENT_TYPE_HTML)) {
```

Comment:
This condition can now be skipped, as you hit `$viewModel()` anyway if it isn't json.